nncf/experimental/torch/fx/transformations.py

```diff
@@ -529,9 +583,217 @@ def fuse_conv_bn(model: torch.fx.GraphModule) -> None:
+    weight scale zp
+        \   |    /
+         Quantize
+            |
+        Dequantize
+
+    and replace with the following:
+     weight scale
+       \     /
+       Multiply
+
+    :param model: Model to apply transformations to.
+    """
+
+    def match_filters(match, original_graph, graph):
+        for node in match.nodes_map:
+            if node.name == "weight" and match.nodes_map[node].op == "get_attr":
+                return True
+        return False
+
+    pattern, replacement = _get_pattern_replacement_per_channel()
+    matches_per_channel = torch.fx.subgraph_rewriter.replace_pattern_with_filters(
+        model, pattern, replacement, [match_filters]
+    )
+    _compress_qdq_constant_transformation(model, matches_per_channel)
+    _set_meta_for_matches(model, matches_per_channel)
+    pattern, replacement = _get_pattern_replacement_per_tensor()
+    matches_per_tensor = torch.fx.subgraph_rewriter.replace_pattern_with_filters(
+        model, pattern, replacement, [match_filters]
+    )
+    _compress_qdq_constant_transformation(model, matches_per_tensor)
+    _set_meta_for_matches(model.graph, matches_per_tensor)
+
+
+def _get_node_inputs(node: torch.fx.Node, model: torch.fx.GraphModule) -> Optional[Tuple[Union[torch.Tensor, int]]]:
+    """
+    Gets the inputs for the Quantize node which quantize the weights. Otherwise returns None.
+
+    :param node: Node to get the inputs from
+    :param model: Model to which the node belongs
+    """
+    args = []
+    for arg in node.args:
+        if isinstance(arg, torch.fx.Node):
+            if arg.op != "get_attr":
+                return None
+            arg = get_tensor_constant_from_node(arg, model)
+        args.append(arg)
+    return tuple(args)
+
+
+def _compress_qdq_constant_transformation(model: torch.fx.GraphModule, matches) -> None:
+    """
+    Change the FP32 weight value to Int8 and also reshape the scale for per_channel_quantization.
+
+    :param: model: Model to apply transformations to.
+    """
+    for match in matches:
+        mul_node = match.replacements[0]
+        sub_node = match.replacements[1]
+        weight_node, scale_node, zp_node, axis = None, None, None, None
+        nodes_map = {node.name: match.nodes_map[node] for node in match.nodes_map}
+        get_const = partial(get_tensor_constant_from_node, model=model)
+        weight_node = get_const(nodes_map["weight"])
+        scale_node = get_const(nodes_map["scale"])
+        zp_node = get_const(nodes_map["zero_point"])
+        axis = nodes_map["axis"]
+        port_id = 0
+        result = None
+        if axis is not None:
+            result = torch.ops.quantized_decomposed.quantize_per_channel.default(
+                weight_node, scale_node, zp_node, axis, -128, 127, torch.int8
+            )
+            _reshape_scale_zp(model, sub_node, mul_node, weight_node, scale_node, zp_node, axis)
```

Comment:
Let's keep qdq creation in scope of one if else statement
```suggestion
        if axis is not None:
            result = torch.ops.quantized_decomposed.quantize_per_channel.default(
                weight_node, scale_node, zp_node, axis, -128, 127, torch.int8
            )
            _reshape_scale_zp(model, sub_node, mul_node, weight_node, scale_node, zp_node, axis)
        else:
            torch.ops.quantized_decomposed.quantize_per_tensor.default(
                weight_node, scale_node, zp_node, -128, 127, torch.int8
            )
            ...
```

Reply:
Alright!

Reply:
Done